nidmresults/objects/generic.py

```diff
@@ -161,6 +161,8 @@ def __init__(self, coordinate_system, nifti_file=None, vox_to_world=None,
 
         self.voxel_to_world = vox_to_world
         self.voxel_size = vox_size
+        if type(dimensions[1])==np.int64:
```

Comment:
@TomMaullin: do you remember why this statement was needed and why overwriting `dimensions`? `int64` seems quite restrictive and I would like to check if there is an alternative solution. 

Apart from that, everything looks good to me and I think we should merge once the tests are okay at https://github.com/incf-nidash/nidmresults-fsl/pull/124.

Reply:
I had a lot of trouble with this... I can't remember which way around this was but there was some discrepancy in the type of `dimensions`.

I can't remember which way around it was but, dimensions is either read in as a int tuple or an numpy.int64 numpy array depending on whether you are running the importer or exporter tests. If I remember correctly, one of these data types came straight from a nibabel function and the other came straight from a JSON reader so I had no choice but to try and account for both.

This if statement checking whether it was numpy int64 is here as even if I converted the tuple to a numpy array (or vice versa) the tests would still error as the type was numpy int64 and not normal int64 or (vice versa). If the data is a numpy type integer the only way I could account for this was by converting them to a normal integer type using the function `.item()` as I have on the line below.

Reply:
Okay... Thanks! Let me try this out and see if I can do anything about it.